test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	plugin "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin"
+)
+
+type pluginConfig struct {
+	cdiDir                 string
+	driverName             string
+	draAddress             string
+	pluginRegistrationPath string
+}
+
+type examplePlugin struct {
+	config             pluginConfig
+	resourceHandleKeys []string
+}
+
+var resourceHandleKeys = []string{"some", "another"}
```

Comment:
What is this? It looks like you expect these to be the only valid keys. Where did that idea come from?

The README says that users can specify arbitrary keys.

Reply:
You said "use marshal" and "validate". That led it. I will remove it.